provider/cluster/manager.go

```diff
@@ -239,13 +239,14 @@ loop:
 		dm.log.Debug("read from runch during shutdown")
 	}
 
-	dm.log.Debug("waiting on dm.wg")
-	dm.wg.Wait()
-
 	if nil != dm.withdrawal {
 		dm.log.Debug("waiting on withdrawal")
-		dm.withdrawal.lc.Shutdown(nil)
+		dm.withdrawal.lc.Done()
```

Comment:
Doesn't this return a channel? What 's the point of calling it here if we don't use the channel for something?

Reply:
good catch!

Reply:
you can use `ShutdownAsync()` if you don't want to wait.  `Done()` does just return a channel that is readable when the object is done.

Reply:
ideally the `withdrawal` object would have `Shutdown()`, `Done()` methods & you can wait on `<-dm.withdrawal.Done()`.

Reply:
it supposed to be `<-dm.withdrawal.lc.Done()`. i just missed read part